model/reviews.js

```diff
@@ -1,63 +1,65 @@
 const db = require("../db/connection");
 
-exports.selectReviewById = (review_id) => {
-  return db
-    .query(
-      `SELECT reviews.*, COUNT(comments.review_id) AS comment_count
+exports.selectReviewById = async (review_id) => {
+  const reviewQueryStr = `SELECT reviews.*, COUNT(comments.review_id) AS comment_count
       FROM reviews
       LEFT JOIN comments
       ON comments.review_id = reviews.review_id
       WHERE reviews.review_id = $1
-      GROUP BY reviews.review_id`,
-      [review_id]
-    )
-    .then(({ rows }) => {
-      if (!rows.length) {
-        return Promise.reject({
-          status: 404,
-          message: `review with id: ${review_id} does not exist`,
-        });
-      }
-      return rows[0];
+      GROUP BY reviews.review_id`;
+  const reviewValue = [review_id];
+
+  const { rows } = await db.query(reviewQueryStr, reviewValue);
+
+  if (!rows.length) {
+    return Promise.reject({
+      status: 404,
+      message: `review with id: ${review_id} does not exist`,
     });
+  }
+  return rows[0];
 };
 
-exports.updateReviewById = (review_id, { inc_votes } = 0) => {
-  return db
-    .query(
-      `UPDATE reviews
+exports.updateReviewById = async (review_id, { inc_votes } = 0) => {
+  const reviewQueryStr = `
+    UPDATE reviews
     SET votes = votes + $1
     WHERE review_id = $2
-    RETURNING *`,
-      [inc_votes, review_id]
-    )
-    .then(({ rows }) => {
-      if (!rows.length) {
-        return Promise.reject({
-          status: 404,
-          message: `review with id: ${review_id} does not exist`,
-        });
-      }
-      return rows[0];
+    RETURNING *`;
+  const reviewValue = [inc_votes, review_id];
+
+  const { rows } = await db.query(reviewQueryStr, reviewValue);
+
+  if (!rows.length) {
+    return Promise.reject({
+      status: 404,
+      message: `review with id: ${review_id} does not exist`,
     });
+  }
+  return rows[0];
 };
 
-exports.selectAllReviews = async () => {
-  const reviewsQueryStr = `
+exports.selectAllReviews = async ({ sort_by = "created_at" }) => {
```

Comment:
The orderBy query should be passed in along with the sortBy one. That way, a user can select, say, the title, and then go into further detail on whether they want it to be ASC or DESC order. By setting this to a default, we can ensure that there's always an order specified, even if the user doesn't set one.

Reply:
Same with category